### PR TITLE
feat(eslint-plugin): added new rule use-default-type-parameter

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Adds the equivalent of TSLint's `use-default-type-parameter` rule.

I'm not sure what to do about the lack of test coverage on the files, since most of them are for edge cases in the TypeScript API not finding information. Would you prefer nodes of unknown types be added as test cases? Will wait for further instructions...

### CODE REVIEW

File: packages/eslint-plugin/src/rules/use-default-type-parameter.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    function checkTSArgsAndParameters(
+      esParameters: TSESTree.TSTypeParameterInstantiation,
+      { typeArguments, typeParameters }: ArgsAndParams,
+    ): void {
+      // Just check the last one. Must specify previous type parameters if the last one is specified.
+      const i = typeArguments.length - 1;
+      const arg = typeArguments[i];
+      const param = typeParameters[i];
+
+      // TODO: would like checker.areTypesEquivalent. https://github.com/Microsoft/TypeScript/issues/13502
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/use-default-type-parameter.ts]: could we just introduce https://github.com/runem/ts-simple-type#readme ?

File: packages/eslint-plugin/src/rules/use-default-type-parameter.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  const sigDecl = sig === undefined ? undefined : sig.getDeclaration();
+  if (sigDecl === undefined) {
+    return ts.isNewExpression(node)
+      ? getTypeParametersFromType(node.expression, checker)
+      : undefined;
+  }
+
+  return sigDecl.typeParameters === undefined
+    ? undefined
+    : sigDecl.typeParameters;
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/use-default-type-parameter.ts]: isn't this ternary useless?

File: packages/eslint-plugin/src/rules/use-default-type-parameter.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/src/rules/use-default-type-parameter.ts]: In theory, sure. Would you like me to do this here or in a separate PR?
